onnxruntime/core/providers/mkldnn/nn/conv.cc

```diff
@@ -402,20 +402,30 @@ Status Conv<T>::Compute(OpKernelContext* context) const {
       src_data = static_cast<T*>(dst.get_data_handle());
     }
 
-    // Reorder filter memory layout if necessary.
-    if (filter_format != conv_primitive->GetFilterMemoryFormat()) {
-      auto pd = mkldnn::memory::primitive_desc(mkldnn::memory::desc(filter_dims_mkl,
-                                                                    MklDnnType<T>(),
-                                                                    filter_format),
-                                               cpu_engine);
-      mkldnn::memory src = mkldnn::memory(pd, (void*)filter_data);
-      // allocate the size queried from memory primitive desc. it may not match tensor logical size due to
-      // mkldnn using padding to allow use of blocked format.
-      filter_reorder_buffer = IAllocator::MakeUniquePtr<void>(alloc, conv_primitive->GetFilterSize());
-      mkldnn::memory dst = mkldnn::memory(conv_fwd_pd->weights_primitive_desc(), filter_reorder_buffer.get());
-      MemoryReorderParams params(src, dst);
-      DoReorder<T>(params);
-      filter_data = static_cast<T*>(dst.get_data_handle());
+    // Reorder filter memory layout if necessary
+    // Avoid data reordering. Save filter memory in mkldnn format from first iteration 
+    // in execution provider mapped by weight name.
+    std::string weightName = OpKernel::Node().InputDefs()[1]->Name();
+    std::shared_ptr<mkldnn::memory> filter_dst_mem = nullptr;
+    if(provider_ != nullptr)
```

Comment:
why do we need all these nullptr checks for the provider?

Reply:
I didn't have this check on my first commit. If for some reason, _provider is null, it won't break. It will continue with reorder.

Reply:
when will the provider be null?
we can add an ONNXRUNTIME_ENFORCE check in the constructor rather than doing all these null checks in the Compute()

Reply:
Thanks. Will remove the check. thanks.